### DISCUSSION TITLE
off-normal sources with periodic boundary conditions

### DISCUSSION THREAD

**Discussion by rcoridan** [NanoComp/meep#2293] [Q&A] [answered]
I have a simple question but the answer is not obvious to me -- If I want to simulate absorption in a simple planar solar cell, I define a source as a plane above the cell over the whole slice through the simulation volume.  The cell is large, so I use periodic boundary conditions.  This is fine for a source where the surface normal of the cell and the plane defining the source are parallel (both aligned along Z, for example).  But how would I simulate an off-normal source?  The periodic boundary conditions would make this difficult.  Is there a way to do this in Meep? 

thanks for any advice.

**Comment by dalarev** [comment, marked as answer]
Have you looked at this [tutorial example](https://meep.readthedocs.io/en/latest/Python_Tutorials/Mode_Decomposition/#reflectance-and-transmittance-spectra-for-planewave-at-oblique-incidence)?

Additionally, take a look at [these test scripts](https://github.com/dalarev/MEEP_scripts/blob/main/test_offnormal_source.ipynb) experimenting with off-normal sources to see if that helps you.

**Reply by rcoridan** [reply]
Thanks for pointing these out to me. I will work through them.

**Comment by stevengj** [comment]
For the theory, see section 4.5 of [this review](https://arxiv.org/pdf/1301.5366.pdf).

**Reply by rcoridan** [reply]
Thanks for attaching this.  I think it will be helpful.

**Comment by oskooi** [comment]
As an additional reference, I reviewed the topic of oblique broadband sources in my [MeepCon tutorial](https://www.youtube.com/watch?v=JE6OfH-tg2w) starting at around the 29th minute.
